core/src/network/handler/mod.rs

```diff
@@ -253,53 +262,73 @@ fn get_content_aspect(
     entry_address: &Address,
     context: Arc<Context>,
 ) -> Result<EntryAspect, HolochainError> {
-    let state = context.state()
-        .ok_or_else(|| {
-            HolochainError::InitializationFailed(
-                String::from("In get_content_aspect: no state found")
-            )
-        })?;
+    let state = context.state().ok_or_else(|| {
+        HolochainError::InitializationFailed(String::from("In get_content_aspect: no state found"))
+    })?;
 
     // Optimistically look for entry in chain...
-    let maybe_chain_header = state.agent()
+    let maybe_chain_header = state
+        .agent()
         .iter_chain()
-        .find(|ref chain_header| chain_header.entry_address() == entry_address);
+        .find(|ref chain_header| chain_header.address() == *entry_address)
+        .map(|h| (h, true))
+        .or_else(|| {
+            state
+                .agent()
+                .iter_chain()
+                .find(|ref chain_header| chain_header.entry_address() == entry_address)
```

Comment:
It might be nice to add a comment or something to point out that this is looking for `entry_address` not `address`. I had to squint for a while to figure out the difference between these two finds.

Reply:
Yeah I'm a tiny bit confused about this part in general it might need a few explainer comments. @maackle what is the idea behind searching both using the header address and the headers entry address?